### PR TITLE
Updated the example of the async void event handler

### LINKED ISSUE #7038: Your example code appears to be wrong.
Reading your example code for the void return type, I'm not sure it does what you think.  If you add that to a console app and run it, you get wildly inaccurate results.  Resetting the counter in the 'event handler' (while declaring an actual event handler that has nothing to do with the code) is just confusing.  I would revisit it with a better example. It doesn't clearly show what you think it does.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: d055727b-c97c-4d25-39b8-a4d3f21d0444
* Version Independent ID: c8a7c6f1-8ee1-5030-db12-3db65a968501
* Content: [Async Return Types (C#)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/async-return-types)
* Content Source: [docs/csharp/programming-guide/concepts/async/async-return-types.md](https://github.com/dotnet/docs/blob/master/docs/csharp/programming-guide/concepts/async/async-return-types.md)
* Product: **dotnet-csharp**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

### PR DESCRIPTION
**Pull request by pkulikov**
Main change is in the dotnet/samples#257

Fixes #7038

### CODE REVIEW

**Comment by rpetrusha** [comment]
Closing and reopening to begin build after PR in samples repo merged.

File: docs/csharp/programming-guide/concepts/async/async-return-types.md [hunk 1]
```diff
@@ -56,7 +56,7 @@ The caller of a void-returning async method can't catch exceptions that are thro
   
 For more information about how to catch exceptions in async methods, see the [Exceptions in Async Methods](../../../language-reference/keywords/try-catch.md#exceptions-in-async-methods) section of the [try-catch](../../../language-reference/keywords/try-catch.md) topic.  
   
-The following eample defines an async event handler.  
+The following example shows the behavior of an async event handler. Note that in the example code an async event handler must let the main thread know when it finishes. Then, the main thread can wait for an async event handler to complete before exiting the program.
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/concepts/async/async-return-types.md]: comma after "example code"
no comma after "Then"

File: docs/csharp/programming-guide/concepts/async/async-return-types.md [hunk 2]
```diff
@@ -56,7 +56,7 @@ The caller of a void-returning async method can't catch exceptions that are thro
...
  
 [!code-csharp[return-value](../../../../../samples/snippets/csharp/programming-guide/async/async-returns3.cs)]
```
Comment (rpetrusha) [inline docs/csharp/programming-guide/concepts/async/async-return-types.md]: I apologize for not looking at the code more carefully when I approved dotnet/samples#257, @pkulikov. There are three typos in it ("Hanlder" --> "Handler"), which are also reflected in the output.

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thanks for submitting this PR, @pkulikov. It looks good, except for two minor comments. There are also some typos in the source code.

**Comment by pkulikov** [comment]
@rpetrusha thank you for review and catching the typo in the source code. dotnet/samples#261 updates the source code and my last commit here has moved a comma to the correct position.

**Comment by mairaw** [comment]
Reopened to grab the latest changes from dotnet/samples

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks, @pkulikov. This looks great. I'll merge now.

**Pull request merged**